api/src/main/java/org/apache/iceberg/expressions/Expressions.java

```diff
@@ -35,8 +37,8 @@ private Expressions() {
   }
 
   public static Expression and(Expression left, Expression right) {
-    Preconditions.checkNotNull(left, "Left expression cannot be null.");
-    Preconditions.checkNotNull(right, "Right expression cannot be null.");
+    requireNonNull(left, "Left expression cannot be null.");
+    requireNonNull(right, "Right expression cannot be null.");
```

Comment:
There's no need to change the existing check, since it already throws NPE. While it may be nice to use Java-standard methods, I don't think it is worth going through and converting all of these cases. That's unnecessary code churn.